generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -22,13 +22,42 @@ def __init__(self, grpc_options):
     def _invoke(self, func, request, metadata=None):
         try:
             response = func(request, metadata=metadata)
-            error_code = response.status
-            error_message = ''
-        #Todo: AB#2325876: The error handling logic will be updated.
         except grpc.RpcError as rpc_error:
-            raise rpc_error
+            error_code = -1
+            error_message = rpc_error.details()
+            samps_per_chan_read = None
+            samps_per_chan_written = None
+            for entry in rpc_error.trailing_metadata() or []:
+                if entry.key == 'ni-error':
+                    value = entry.value if isinstance(entry.value, str) else entry.value.decode('utf-8')
+                    try:
+                        error_code = int(value)
+                    except ValueError:
+                        error_message += f'\nError status: {value}'
+                if entry.key == "samps_per_chan_read":
+                    samps_per_chan_read = entry.value
+                if entry.key == "samps_per_chan_written":
+                    samps_per_chan_written = entry.value
+            grpc_error = rpc_error.code()
+            if grpc_error == grpc.StatusCode.UNAVAILABLE:
+                error_message = 'Failed to connect to server'
+            elif grpc_error == grpc.StatusCode.UNIMPLEMENTED:
+                error_message = (
+                    'This operation is not supported by the NI gRPC Device Server being used. Upgrade NI gRPC Device Server.'
+                )
+            self.raise_error(error_code, error_message, samps_per_chan_written, samps_per_chan_read)
         return response
 
+    def raise_error(self, error_code, error_message, samps_per_chan_written=None, samps_per_chan_read=None):
+        if error_code < 0:
```

Comment:
What is the expectations for the errors whose error code are greater than 0? How are we supposed to handle it?

Reply:
errors > 0 are a warning. We use the `warnings` module for that.

Reply:
I have created a `DriverWarning` class to handle all warnings in DAQmx. But there already exists a `DaqWarning` class in `errors.py`, so should we try to reuse it or can we stick with the newly added warning class. I am not sure where this `DaqWarning` is used.